### ISSUE TITLE
[Bug] Get Permission error when try to downgrade an app

### ISSUE THREAD

**Issue by MrEngineerMind** [NeoApplications/Neo-Backup#633]
**Description**
When I try to restore a lower version of an app using a Neo backup file, it gives an error.

**Steps To Reproduce**

1. I have "Allow downgrading" enabled and "Restore with all permissions" disabled for security reasons.
2. I did a backup of a simple game app using Neo 8.1.1.
3. I installed the updated version of that app from the play store and did a backup of it using NB 8.1.1
4. I then tried to restore the previous version of the app using NB 8.1.1
5. NB displayed the following

**Expected behavior**
For it to restore the previous version of the app and data for the app and it runs fine

**Screenshots**
![Downgrade](https://user-images.githubusercontent.com/18239066/185523608-82b89209-193c-48be-a084-328000472ea2.png)

**System Information(please complete the following information):**

- Device: Pixel 4a
- Android Version:  10
- ROM: (AOSP, MIUI, OOS...)  Stock
- App's Version: 8.1.1

**Comment by whitedavidp** [comment]
Sorry. Did the restore actually happen (but with permissions not granted) or did it completely fail?

**Comment by MrEngineerMind** [comment]
This error should be ignored if possible because the fact that you are "down-grading" a version of an app sort of assumes the older version (that you are trying to downgrade to) will possible lack certain functionality vs. the more recent version that is being overwritten.

And the app is a simple game that I do not give it any permissions to for either the old or new version.

**Comment by hg42** [comment]
> No, the newer version stayed installed (after I tried to downgrade it and then got the above error)

that's actually interesting, because I first thought, the error would orginate from restoring the permissions (which would be `pm grant` command), but the error clearly says it's from the `pm install` command.

So, I have to think again...

There are two variants where we explicitly restore permissions:
* **all** permissions are granted by using `-g` on the `pm install` command
* **individual** permissions are granted with `pm grant` commands

in your case, it's none of them

so the error is talking about something different.
I guess it's because the permissions are currently granted to the app, but the installed app is for an older sdk, so the pm install cannot migrate the permissions.

So, probably, the solution should be to uninstall the apk first and then install the older apk and data.

**Comment by MrEngineerMind** [comment]
Yes, uninstalling the current version and using NB to install the older version worked.
